rascaline/src/math/splines.rs

```diff
@@ -9,21 +9,21 @@ const MAX_SPLINE_SIZE: usize = 10_000;
 
 /// [Hermit cubit spline][splines-wiki] implementation.
 ///
-/// This kind of spline uses information of the value of a function on control
-/// points as well as the gradient of the function at these points. This
+/// This kind of spline uses information of the values of a function on control
+/// points as well as the derivatives of the function at these points. This
 /// implementation takes a single scalar as input, and output array values, i.e.
 /// we can spline functions of the form `R -> R^n`.
 ///
 /// [splines-wiki]: https://en.wikipedia.org/wiki/Cubic_Hermite_spline
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, serde::Serialize)]
```

Comment:
I would not put `serde::Serialize` here, since it would force us to increase the version number if we ever change the names/types of the fields. For the test adding a `pub(crate) fn points(&self) -> &[HermitSplinePoint<D]` should be enough to then extract positions/values/derivatives and serialize the whole thing.

Reply:
You want to add this function here? 

Isn't it then just enough to change the `points` below to `pub(crate) points`?

Reply:
yeah, that works as well ^^